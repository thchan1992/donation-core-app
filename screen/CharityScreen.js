import { useState } from "react";
import { View, Text, StyleSheet, Image, ScrollView } from "react-native";
import PrimaryButton from "../components/PrimaryButton";
import { horizontalLine } from "../util/styling";
const CharityScreen = ({ navigation, route }) => {
  const [showDet, setShowDet] = useState(false);
  const { charity } = route.params;
  return (
    <View style={styles.container}>
      {!showDet && <Image style={styles.image} source={charity.imageUrl} />}
      <Text style={styles.text}>{charity.name}</Text>
      <View
        style={[
          { flexDirection: "row" },
          { alignItems: "center" },
          showDet ? horizontalLine.horizontalLine : undefined,
        ]}
      >
        <PrimaryButton
          text={"Donate"}
          onPress={() => {
            navigation.navigate("GetAmountScreen", { charity: charity });
          }}
        />
        {!showDet && (
          <PrimaryButton
            text={"Detail"}
            onPress={() => {
              setShowDet(true);
            }}
          />
        )}
        {showDet && (
          <PrimaryButton
            text={"Hide"}
            onPress={() => {
              setShowDet(false);
            }}
          />
        )}
      </View>
      {showDet && (
        <ScrollView>
          <Text style={styles.infoText}>{charity.info}</Text>
        </ScrollView>
      )}
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: "center",
    paddingTop: 10,
  },
  image: {
    paddingTop: 40,
    width: 250,
    height: 250,
    borderWidth: 3,
    borderColor: "#FF304F",
    borderRadius: 8,
    padding: 30,
  },
  text: {
    alignSelf: "center",
    padding: 3,
    fontWeight: "bold",
    textTransform: "uppercase",
    fontSize: 20,
    color: "white",
  },
  infoText: {
    padding: 10,
    fontSize: 30,
    color: "white",
  },
});

export default CharityScreen;
