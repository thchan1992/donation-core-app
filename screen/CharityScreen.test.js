import React from "react";
import renderer from "react-test-renderer";
import CharityScreen from "./CharityScreen";
import { act } from "react-test-renderer";

// describe("<CharityScreen />", () => {
//   const navigation = { navigate: jest.fn() };
//   const route = {
//     params: {
//       charity: { name: "charity 1", imageUrl: "image.com", info: "info" },
//     },
//   };
//   it("has 3 child", () => {
//     const tree = renderer
//       .create(<CharityScreen navigation={navigation} route={route} />)
//       .toJSON();
//     expect(tree.children.length).toBe(3);
//   });
// });

describe("CharityScreen", () => {
  const navigation = { navigate: jest.fn() };
  const route = {
    params: {
      charity: { name: "charity 1", imageUrl: "image.com", info: "info" },
    },
  };
  let component;
  beforeEach(() => {
    component = renderer.create(
      <CharityScreen navigation={navigation} route={route} />
    );
  });

  it("should match the snapshot", () => {
    expect(component.toJSON()).toMatchSnapshot();
  });

  it("should call navigation.navigate when donate button is pressed", () => {
    const instance = component.root;
    const donateButton = instance.findByProps({ text: "Donate" });
    expect(donateButton).toBeTruthy();
    act(() => {
      donateButton.props.onPress();
    });
    expect(navigation.navigate).toHaveBeenCalledWith("GetAmountScreen", {
      charity: expect.any(Object),
    });
  });

});
