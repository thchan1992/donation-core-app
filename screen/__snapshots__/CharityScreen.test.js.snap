// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharityScreen renders correctly with default state 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "paddingTop": 10,
    }
  }
>
  <Image
    source="image.com"
    style={
      Object {
        "borderColor": "#FF304F",
        "borderRadius": 8,
        "borderWidth": 3,
        "height": 300,
        "padding": 30,
        "paddingTop": 40,
        "width": 300,
      }
    }
  />
  <Text
    style={
      Object {
        "alignSelf": "center",
        "color": "white",
        "fontSize": 20,
        "fontWeight": "bold",
        "padding": 3,
        "textTransform": "uppercase",
      }
    }
  >
    charity 1
  </Text>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
        },
        Object {
          "alignItems": "center",
        },
        undefined,
      ]
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#28C7FA",
          "borderColor": "#FF304F",
          "borderRadius": 8,
          "borderWidth": 1.5,
          "height": 70,
          "margin": 10,
          "opacity": 1,
          "width": 150,
        }
      }
    >
      <Text
        style={
          Object {
            "alignSelf": "center",
            "fontSize": 30,
            "fontWeight": "bold",
            "padding": 16,
          }
        }
      >
        Donate
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#28C7FA",
          "borderColor": "#FF304F",
          "borderRadius": 8,
          "borderWidth": 1.5,
          "height": 70,
          "margin": 10,
          "opacity": 1,
          "width": 150,
        }
      }
    >
      <Text
        style={
          Object {
            "alignSelf": "center",
            "fontSize": 30,
            "fontWeight": "bold",
            "padding": 16,
          }
        }
      >
        Detail
      </Text>
    </View>
  </View>
</View>
`;

exports[`CharityScreen renders correctly with showDet state true 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "paddingTop": 10,
    }
  }
>
  <Image
    source="image.com"
    style={
      Object {
        "borderColor": "#FF304F",
        "borderRadius": 8,
        "borderWidth": 3,
        "height": 300,
        "padding": 30,
        "paddingTop": 40,
        "width": 300,
      }
    }
  />
  <Text
    style={
      Object {
        "alignSelf": "center",
        "color": "white",
        "fontSize": 20,
        "fontWeight": "bold",
        "padding": 3,
        "textTransform": "uppercase",
      }
    }
  >
    charity 1
  </Text>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
        },
        Object {
          "alignItems": "center",
        },
        undefined,
      ]
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#28C7FA",
          "borderColor": "#FF304F",
          "borderRadius": 8,
          "borderWidth": 1.5,
          "height": 70,
          "margin": 10,
          "opacity": 1,
          "width": 150,
        }
      }
    >
      <Text
        style={
          Object {
            "alignSelf": "center",
            "fontSize": 30,
            "fontWeight": "bold",
            "padding": 16,
          }
        }
      >
        Donate
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#28C7FA",
          "borderColor": "#FF304F",
          "borderRadius": 8,
          "borderWidth": 1.5,
          "height": 70,
          "margin": 10,
          "opacity": 1,
          "width": 150,
        }
      }
    >
      <Text
        style={
          Object {
            "alignSelf": "center",
            "fontSize": 30,
            "fontWeight": "bold",
            "padding": 16,
          }
        }
      >
        Detail
      </Text>
    </View>
  </View>
</View>
`;

exports[`CharityScreen should match the snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "paddingTop": 10,
    }
  }
>
  <Image
    source="image.com"
    style={
      Object {
        "borderColor": "#FF304F",
        "borderRadius": 8,
        "borderWidth": 3,
        "height": 300,
        "padding": 30,
        "paddingTop": 40,
        "width": 300,
      }
    }
  />
  <Text
    style={
      Object {
        "alignSelf": "center",
        "color": "white",
        "fontSize": 20,
        "fontWeight": "bold",
        "padding": 3,
        "textTransform": "uppercase",
      }
    }
  >
    charity 1
  </Text>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
        },
        Object {
          "alignItems": "center",
        },
        undefined,
      ]
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#28C7FA",
          "borderColor": "#FF304F",
          "borderRadius": 8,
          "borderWidth": 1.5,
          "height": 70,
          "margin": 10,
          "opacity": 1,
          "width": 150,
        }
      }
    >
      <Text
        style={
          Object {
            "alignSelf": "center",
            "fontSize": 30,
            "fontWeight": "bold",
            "padding": 16,
          }
        }
      >
        Donate
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#28C7FA",
          "borderColor": "#FF304F",
          "borderRadius": 8,
          "borderWidth": 1.5,
          "height": 70,
          "margin": 10,
          "opacity": 1,
          "width": 150,
        }
      }
    >
      <Text
        style={
          Object {
            "alignSelf": "center",
            "fontSize": 30,
            "fontWeight": "bold",
            "padding": 16,
          }
        }
      >
        Detail
      </Text>
    </View>
  </View>
</View>
`;

exports[`CharityScreen should show details when detail button is pressed 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "paddingTop": 10,
    }
  }
>
  <Text
    style={
      Object {
        "alignSelf": "center",
        "color": "white",
        "fontSize": 20,
        "fontWeight": "bold",
        "padding": 3,
        "textTransform": "uppercase",
      }
    }
  >
    charity 1
  </Text>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
        },
        Object {
          "alignItems": "center",
        },
        Object {
          "borderBottomColor": "#FF304F",
          "borderBottomWidth": 0.5,
        },
      ]
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#28C7FA",
          "borderColor": "#FF304F",
          "borderRadius": 8,
          "borderWidth": 1.5,
          "height": 70,
          "margin": 10,
          "opacity": 1,
          "width": 150,
        }
      }
    >
      <Text
        style={
          Object {
            "alignSelf": "center",
            "fontSize": 30,
            "fontWeight": "bold",
            "padding": 16,
          }
        }
      >
        Donate
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#28C7FA",
          "borderColor": "#FF304F",
          "borderRadius": 8,
          "borderWidth": 1.5,
          "height": 70,
          "margin": 10,
          "opacity": 1,
          "width": 150,
        }
      }
    >
      <Text
        style={
          Object {
            "alignSelf": "center",
            "fontSize": 30,
            "fontWeight": "bold",
            "padding": 16,
          }
        }
      >
        Hide
      </Text>
    </View>
  </View>
  <RCTScrollView>
    <View>
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 30,
            "padding": 10,
          }
        }
      >
        info
      </Text>
    </View>
  </RCTScrollView>
</View>
`;
